### DISCUSSION TITLE
how can I perfectly connect two waveguides?

### DISCUSSION THREAD

**Discussion by wuzhonghan** [gdsfactory/gdsfactory#2422] [Q&A] [answered]
if I rotate a straight waveguide and connect another waveguide two it, the connection is not perfect, can the function "connect" be modified to fixed this problem?

[code]
import gdsfactory as gf

c = gf.Component()

wg1_ref = c << gf.components.straight(length=50)
wg2_ref = c << gf.components.straight(length=50)

wg1_ref.rotate(30)
wg2_ref.connect('o1',destination=wg1_ref.ports['o2'])

c.show()

![image](https://github.com/gdsfactory/gdsfactory/assets/124678531/92a18f62-955e-415b-8b07-1e429246379d)
![image](https://github.com/gdsfactory/gdsfactory/assets/124678531/8e636285-3478-4f57-b691-54d6727877d2)

**Comment by joamatab** [comment, marked as answer]
Hi,

see tutorial for non manhattan connections
https://gdsfactory.github.io/gdsfactory/notebooks/04_routing_non_manhattan.html

You can do this

```python

import gdsfactory as gf

c = gf.Component()

wg1_ref = c << gf.components.straight(length=50)
wg2_ref = c << gf.components.straight(length=50)

wg1_ref.rotate(30)
wg2_ref.connect('o1',destination=wg1_ref.ports['o2'])
c = c.flatten_invalid_refs()
c.show()

```

![image](https://github.com/gdsfactory/gdsfactory/assets/4514346/021eab70-9bb9-4a18-beef-077626c9945a)

**Reply by wuzhonghan** [reply]
Thanks!
